file_transfer_api/src/file_manager/aws_file_manager.py

```diff
@@ -0,0 +1,110 @@
+from pathlib import Path
+from uuid import uuid4
+
+from src.file_manager.abstract_file_manager import AbstractFileManager
+import boto3
+import os
+from botocore.exceptions import NoCredentialsError
+from src.exceptions.file_exceptions import (FileDownloadError, FileUploadError, FileDeleteError, FileDoesNotExistError,
+                                            FileUpdateError)
+from src.file_manager.utils.url_utils import URLPath
+from typing import IO, Optional
+
+# Get download location from the environment
+download_path = os.getenv("DOWNLOAD_DIRECTORY", default="data/downloads")
+
+
+class AWSFileManager(AbstractFileManager):
+    def __init__(self, target_bucket: str, s3_client: Optional[boto3.client] = None):
```

Comment:
I don't think boto3.client is the right type. https://www.reddit.com/r/learnprogramming/comments/12635xu/type_hinting_with_boto3/

Reply:
Thanks for this. Looks like it's a variant of botocore.client.BaseClient. Have updated in the style of the Reddit article above. Thanks!